File: leetcode-everyday/贪心算法/452-用最少数量的箭射爆气球/452-用最少数量的箭射爆气球.js
/**
 * @param {number[][]} points
 * @return {number}
 */
 var findMinArrowShots = function(points) {
  if(!points.length) return 0;
  if(points.length === 1) return 1;
  points.sort((a, b) => a[0] - b[0])
  let res = 1;
  let prev = points[0][1];
  for(let i = 1; i < points.length; i++) {
      if(prev >= points[i][0]) {
          prev = Math.min(prev, points[i][1]);
      } else {
          res ++;
          prev = points[i][1];
      }
  }
  return res;
};

File: leetcode-everyday/贪心算法/452-用最少数量的箭射爆气球/452-用最少数量的箭射爆气球.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if(points.empty()) return 0;
        if(points.size() == 1) return 1;
        sort(points.begin(), points.end(), [](vector<int>& a, vector<int>& b) {
            return a[0] < b[0];
        });
        int res = 1, prev = points[0][1];
        for(int i = 1; i < points.size(); i++) {
            if(prev >= points[i][0]) {
                prev = min(prev, points[i][1]);
            } else {
                res ++;
                prev = points[i][1];
            }
        }
        return res;
    }
};